syft/frameworks/torch/linalg/lr.py

```diff
@@ -0,0 +1,272 @@
+import torch
+import numpy as np
+from syft.workers import BaseWorker
+from syft.frameworks.torch.linalg.operations import inv_sym
+from syft.generic.pointers.pointer_tensor import PointerTensor
+
+from scipy.stats import t
+from typing import List
+import random
+
+
+class BloomRegressor:
+    """
+    Multi-Party Linear Regressor based on Jonathan Bloom's algorithm.
+    It performs linear regression using Secured Multi-Party Computation
+
+    Reference: Section 2 of https://arxiv.org/abs/1901.09531
+
+    Args:
+        cryto_provider: a BaseWorker providing crypto elements for ASTs such as
+            Beaver triples
+        hbc_worker: The "Honest but Curious" BaseWorker
+        precision_fractional: precision chosen for FixedPrecisionTensors
+        fit_intercept:  whether to calculate the intercept for this model. If set
+            to False, no intercept will be used in calculations (e.g. data is
+            expected to be already centered)
+
+    Attributes:
+        coef: torch.Tensor of shape (n_features, ). Estimated coefficients for
+            the linear regression problem.
+        intercept: torch.Tensor of shape (1, ) if fit_intercept is set to True,
+            None otherwise. Estimated intercept for the linear regression.
+        pvalue_coef: numpy.array of shape (n_features, ). Two-sided p-value for a
+            hypothesis test whose null hypothesis is that the each coeff is zero.
+        pvalue_intercept: numpy.array of shape (1, ) if fit_intercept is set to True,
+            None otherwise. Two-sided p-value for a hypothesis test whose null
+            hypothesis is that the intercept is zero.
+
+    """
+
+    def __init__(
+        self,
+        crypto_provider: BaseWorker,
+        hbc_worker: BaseWorker,
+        precision_fractional: int = 6,
+        fit_intercept=True,
+    ):
+
+        self.crypto_provider = crypto_provider
+        self.hbc_worker = hbc_worker
+        self.precision_fractional = precision_fractional
+        self.fit_intercept = fit_intercept
+
+    def fit(self, X_ptrs: List[torch.Tensor], y_ptrs: List[torch.Tensor]):
+        """
+        Fits the linear model using Secured Multi-Party Linear Regression.
+        The final results (i.e. coefficients and p-values) will be public.
+        """
+
+        # Checking if the pointers are as expected
+        self._check_ptrs(X_ptrs, y_ptrs)
+
+        # Check if each y is a 2-dim or 1-dim tensor, unsqueeze if it is not
```

Comment:
Is this comment correct?

Reply:
Yes, it's a bit ambiguous though...

I meant "Check if each y is a 2-dim or 1-dim tensor, unsqueeze if it 1-dim"